File: Laborator/Lab 4/JavaClientServer/Persistance/src/main/java/Repository/DBRepositoryAdmin.java
package Repository;

import Domain.Admin;
import Utils.DbUtils;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class DBRepositoryAdmin implements IRepositoryAdmin<String, Admin> {
    private DbUtils dbUtils;
    private static final Logger logger = LogManager.getLogger();

    public DBRepositoryAdmin(Properties props) {
        this.dbUtils = new DbUtils(props);
    }

    private void testSelect() {
        logger.traceEntry("Testing select clause...");
        Connection con = dbUtils.getConnection();
        try (PreparedStatement statement = con.prepareStatement("select * from registeredusers")) {
            try (ResultSet result = statement.executeQuery()) {
                if (result.next())
                    do {
                        logger.trace("found username " + result.getString("Name"));
                    } while (result.next());
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB " + ex);
        } finally {
            try {
                con.close();
                logger.info("closed");
            } catch (SQLException ex) {
                logger.error(ex);
            }
        }

    }

    @Override
    public Admin find(String username) {
        logger.traceEntry("finding admin with username {} ", username);
        //testSelect();
        Connection con = dbUtils.getConnection();
        Admin e = null;
        try (PreparedStatement preStmt = con.prepareStatement("select * from registeredusers where Username=?")) {
            preStmt.setString(1, username);
            try (ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    String Username = result.getString("Username");
                    String name = result.getString("Name");
                    String password = result.getString("Password");
                    e = new Admin(Username, name, password);
                    logger.traceExit("Adminul este {}", e);
                }
            }
        } catch (SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB " + ex);
        } finally {
            try {
                con.close();
                logger.info("closed");
            } catch (SQLException ex) {
                logger.error(ex);
            }
        }

        if (e == null)
            logger.traceExit("No admin found with username {}", username);
        return e;
    }
}

File: Laborator/Lab 4/C# Client-server/Persistence/Repository/Database/DbRepositoryAdmin.cs
using Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Persistence.Utils;
using System.Collections.Specialized;
using log4net;
using System.Configuration;
using System.Data;
using MySql.Data.MySqlClient;

namespace Persistence.Repository.Database
{
    public class DbRepositoryAdmin : IRepositoryAdmin<string, Admin>
    {
        private DbUtils dbUtils;
        private static ILog logger = LogManager.GetLogger(typeof(DbRepositoryAdmin));


        public DbRepositoryAdmin(NameValueCollection properties)
        {
            dbUtils = new DbUtils(properties);

            logger.Info("Created new DbRepositoryAdmin instance");
        }
        
        public Admin Find(string username)
        {
            logger.Debug("Finding admin with username " + username);
            MySqlConnection connection = (MySqlConnection)dbUtils.GetConnection();

            MySqlCommand selectCommand = new MySqlCommand("select * from RegisteredUsers where Username=@username", connection);
            selectCommand.Parameters.Add("@username", MySqlDbType.VarChar).Value = username;
            MySqlDataAdapter adapter = new MySqlDataAdapter(selectCommand);
            DataSet dataSet = new DataSet();
            
            adapter.Fill(dataSet);

            if (dataSet.Tables[0].Rows.Count == 0)
            {
                logger.Debug("No admin found with username " + username);
                return null;
            }

            logger.Debug("Successfully found admin with username " + username);

            return new Admin()
            {
                ID = (string)dataSet.Tables[0].Rows[0]["Username"],
                Name = (string)dataSet.Tables[0].Rows[0]["Name"],
                Password = (string)dataSet.Tables[0].Rows[0]["Password"]
            };
        }
    }
}
